## Data Source

Foreign donation data source: https://catalog.data.gov/dataset/foreign-gifts-and-contracts-report-2011

Data description from source:

>"The Foreign Gift and Contracts Report contains all gifts received within the previous six years from any foreign source, contracts with a foreign entity, and/or any ownership interests in or control over the institution by a foreign entity reported by Title IV-eligible domestic institutions that offer a bachelor's degree or higher, or that offer a transfer program of not less than two years that is acceptable for credit toward a bachelor's degree, when the total value of all gifts given in a calendar year is $250,000 or greater or if the institution is owned or controlled by a foreign source."

Data last updated: August 9, 2018

**Note**: All recorded donations dated back 6 years, _i.e._ 2012-2018.

### Initial loading of foreign gifts data from data.gov

In [1]:
%matplotlib widget
import pandas as pd
import seaborn as sns

sns.set()

donations_df = pd.read_excel('./data/ForeignGifts.xls', skiprows=1)
# donations_df.loc[:, 'Foreign Gift Amount'] = donations_df.loc[:, 'Foreign Gift Amount'].apply('{:,}'.format)
donations_df.head(25)

ID   OPEID                       Institution Name          City State  \
0    1  324200             Carnegie Mellon University    Pittsburgh    PA   
1    2  143400              American University (The)    Washington    DC   
2    3  143400              American University (The)    Washington    DC   
3    4  143400              American University (The)    Washington    DC   
4    5  143400              American University (The)    Washington    DC   
5    6  143400              American University (The)    Washington    DC   
6    7  143400              American University (The)    Washington    DC   
7    8  211500                        Amherst College       Amherst    MA   
8    9  108100               Arizona State University         Tempe    AZ   
9   10  108100               Arizona State University         Tempe    AZ   
10  11  108100               Arizona State University         Tempe    AZ   
11  12  108100               Arizona State University         Tempe    AZ   
12  13  108100               Arizona State University         Tempe    AZ   
13  14  108100               Arizona State University         Tempe    AZ   
14  15  324200             Carnegie Mellon University    Pittsburgh    PA   
15  16  229000              Michigan State University  East Lansing    MI   
16  17  137000         University of Colorado Boulder       Boulder    CO   
17  18  217800  Massachusetts Institute of Technology     Cambridge    MA   
18  19  373500       Virginia Commonwealth University      Richmond    VA   
19  20  113100     California Institute of Technology      Pasadena    CA   
20  21  110800                 University of Arkansas  Fayetteville    AR   
21  22  108100               Arizona State University         Tempe    AZ   
22  23  108100               Arizona State University         Tempe    AZ   
23  24  108100               Arizona State University         Tempe    AZ   
24  25  108100               Arizona State University         Tempe    AZ   

   Foreign Gift Received Date  Foreign Gift Amount      Gift Type  \
0                  2015-01-01             77391377       Contract   
1                  2012-01-25               500000  Monetary Gift   
2                  2013-05-28               500000  Monetary Gift   
3                  2015-09-10               500000  Monetary Gift   
4                  2012-05-17               610000  Monetary Gift   
5                  2014-04-24              1000000  Monetary Gift   
6                  2015-04-13              1000000  Monetary Gift   
7                  2018-04-05               500000  Monetary Gift   
8                  2013-03-22                   23       Contract   
9                  2018-03-28                  130       Contract   
10                 2018-06-28                  450       Contract   
11                 2015-12-04                  895       Contract   
12                 2015-11-12                 2580       Contract   
13                 2016-09-06                 5310       Contract   
14                 2017-10-27             67700000       Contract   
15                 2012-12-14             45128974       Contract   
16                 2014-08-01             43021197       Contract   
17                 2014-05-06             43000000  Monetary Gift   
18                 2015-07-01             40092588       Contract   
19                 2016-11-29             40000000  Monetary Gift   
20                 2015-12-31             39108635       Contract   
21                 2017-10-12                 5436       Contract   
22                 2014-01-22                 5763       Contract   
23                 2016-08-04                 5980       Contract   
24                 2016-11-15                 5980       Contract   

       Country of Giftor                          Giftor Name  
0                  QATAR       Qatar Foundation for Education  
1                BAHRAIN                   Kingdom of Bahrain  
2             

## Universities receiving foreign donations  
### Top 10 Universities based on number of foreign donations received

In [2]:
(donations_df
 .groupby('Institution Name')   # Grouping by unisveristy
 .count()                       # Applying count() to group
 .iloc[:, 0]                    # Only display first column
 .sort_values(ascending=False)  # Sort in decending order
 .head(10))                     # Only show top 10

Institution Name
University of California, Los Angeles          3611
California Institute of Technology             3364
Johns Hopkins University                       1181
Columbia University in the City of New York     828
Ohio State University (The)                     795
University of Michigan - Ann Arbor              702
Stanford University                             659
Harvard University                              564
Massachusetts Institute of Technology           415
University of Colorado Denver                   412
Name: ID, dtype: int64

In [3]:
import matplotlib.pyplot as plt


top10_by_counts = (
    donations_df.groupby('Institution Name')
    .count()
    .iloc[:, 0]
    .sort_values(ascending=True)
    .iloc[-10:])

fig, ax = plt.subplots(figsize=(10,5))                                 # Begin pyplot area

ax.barh(top10_by_counts.index, top10_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations received')
ax.set_title('Top 10 universities based on number of foreign donations recieved (2012-2018)')
for i, v in enumerate(top10_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)
plt.tight_layout()
fig.show()
plt.savefig('temp.png')

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

### Top 10 schools based on total dollars of foreign donations received

In [6]:
(donations_df
 .groupby('Institution Name')     # Group by university
 .sum()                           # Apply sum() to all columns
 .loc[:, 'Foreign Gift Amount']   # Only display 'Foresign Gift Amount' columm
 .sort_values(ascending=False)    # Sort in decending
 .head(10))                        # Display top 10 entries only

Institution Name
Carnegie Mellon University               1378003807
Harvard University                       1143899998
Massachusetts Institute of Technology     660830374
Johns Hopkins University                  447459034
Northwestern University                   420550173
Georgetown University                     415622195
University of California, Berkeley        336801721
Duke University                           314693237
Stanford University                       313775195
New York University                       291391003
Name: Foreign Gift Amount, dtype: int64

In [10]:
import dill


top10_schools_all = (donations_df
 .groupby('Institution Name')     # Group by university
 .sum()                           # Apply sum() to all columns
 .loc[:, 'Foreign Gift Amount']   # Only display 'Foresign Gift Amount' columm
 .sort_values(ascending=False)    # Sort in decending
 .head(10))                        # Display top 10 entries only

dill.dump(top10_schools_all, open('./top_10_schools_all.pkd', 'wb'))

In [5]:
top10_by_dollars = (donations_df
    .groupby('Institution Name')
    .sum()
    .loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=True)
    .iloc[-10:])

fig, ax = plt.subplots(figsize=(10,5))                                  # Initialize pyplot area
ax.barh(top10_by_dollars.index, top10_by_dollars.values*(10**-9), color='purple')   # Generating bar chart
ax.set_xlabel('Billion Dollars')
ax.set_title('Top 10 universities based on dollars of foreign donations received (2012-2018)')
for i, v in enumerate(top10_by_dollars):
    ax.text(v*(10**-9), i,'\$'+str('{:.2f}'.format(v*(10**-9))+'B'), color='black')
plt.margins(x=0.15)
plt.tight_layout()
plt.show()

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

### Top 10 schools based on total dollars of foreign donations received (cash only)

In [45]:
(donations_df
 .loc[donations_df['Gift Type']=='Monetary Gift']  # Pullinng all rows that are 'Monetary Gift' only (at 'Gift Type' column)
 .groupby('Institution Name')                      # Group by university
 .sum()                                            # Apply sum() over columns
 .loc[:, 'Foreign Gift Amount']                    # Only display 'Foreign Gift Amount column'
 .sort_values(ascending=False)                     # Sort in descending order
 .head(10))                                        # Display top 10 entries only

Institution Name
Harvard University                             729378154
Massachusetts Institute of Technology          582888432
Johns Hopkins University                       447443471
Stanford University                            305730258
Duke University                                242147330
Columbia University in the City of New York    146003397
New York University                            140389153
University of Pennsylvania                     135399121
University of California, Berkeley             132745273
Yale University                                112274698
Name: Foreign Gift Amount, dtype: int64

In [49]:
top10_by_cash = (donations_df.loc[donations_df['Gift Type']=='Monetary Gift']
 .groupby('Institution Name')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=True)
 .iloc[-10:])

fig, ax = plt.subplots(figsize=(10,5))                                   # Initializing pyplot area
ax.barh(top10_by_cash.index, top10_by_cash.values*(10**-6), color='purple')    # Generating bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Top 10 universities based on dollar of foreign cash donations recieved (2012-2018)')
for i, v in enumerate(top10_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

# Note: this is donated contract + cash gifts

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

## Data Source

Democracy Index data scrapped from: https://en.wikipedia.org/wiki/Democracy_Index, original data from: https://www.eiu.com/public/topical_report.aspx?campaignid=democracyindex2019

Democracy Index description from source: 
>"The EIU Democracy Index provides a snapshot of the state of world democracy for 165 independent states and two territories. The Democracy Index is based on five categories: electoral process and pluralism; civil liberties; the functioning of government; political participation; and political culture. Based on their scores on 60 indicators within these categories, each country is then itself classified as one of four types of regime: full democracy; flawed democracy; hybrid regime; and authoritarian regime."

**Note**: Full Democracy Index report available on repository

### Initial loading of Democracy Index data

**Note**: Data scrapped from Wikipedia, which cited the [EIU](https://www.eiu.com/topic/democracy-index), via the MS Excel "data from web" function.

In [11]:
labels = [                                    # List of column headings for democracy index data
    'Rank',
    'Country',
    'Score',
    'Electoral process and pluralism',
    'Functioning of government',
    'Political participation',
    'Political culture',
    'Civil liberties',
    'Regime type',
    'Region',
    '','','','','']                          # Empty labels needed because of the shape of the data file
democracy_df = pd.read_excel(
    './data/democracy_index.xlsx',
    names=labels,                            # Use list generated above for column labels
    skiprows=148)                            # Number of rows to skip determined by brute force trail and error

democracy_df.drop(
    ['', '.1', '.2', '.3', '.4'],            # Dropping empty columns        
    axis=1,
    inplace=True)
democracy_df = democracy_df.iloc[2:310]      # Dropping superfluous columns to the left
democracy_df.loc[:, 'Country'] = (
    democracy_df.loc[:, 'Country']           # There were extra whitespace to the left of the name of countries 
    .str                                     # Accessing string methods
    .strip())                                # Strip white space on both side
democracy_df = (                             
    democracy_df.loc[democracy_df['Country'] # Slicing out all empty rows
    .isnull() == False])                     # Identifying empty rows by empty cells in 'Country' column    
democracy_df.fillna(                         # Fill in missing rankings
    method='ffill',
    inplace=True)
democracy_df.set_index('Country', inplace=True)
democracy_df.rename(                         # Rename the one country name with superscript notes
    index={'South Korea[n 2]': 'South Korea'},
    inplace=True)
democracy_df.head(20)

Rank  Score  Electoral process and pluralism  \
Country                                                        
Norway             1   9.87                            10.00   
Iceland            2   9.58                            10.00   
Sweden             3   9.39                             9.58   
New Zealand        4   9.26                            10.00   
Finland            5   9.25                            10.00   
Ireland            6   9.24                            10.00   
Denmark            7   9.22                            10.00   
Canada             7   9.22                             9.58   
Australia          9   9.09                            10.00   
Switzerland       10   9.03                             9.58   
Netherlands       11   9.01                             9.58   
Luxembourg        12   8.81                            10.00   
Germany           13   8.68                             9.58   
United Kingdom    14   8.52                             9.58   
Uruguay           15   8.38                            10.00   
Austria           16   8.29                             9.58   
Spain             16   8.29                             9.58   
Mauritius         18   8.22                             9.17   
Costa Rica        19   8.13                             9.58   
France            20   8.12                             9.58   

                Functioning of government  Political participation  \
Country                                                              
Norway                               9.64                    10.00   
Iceland                              9.29                     8.89   
Sweden                               9.64                     8.33   
New Zealand                          9.29                     8.89   
Finland                              8.93                     8.89   
Ireland                              7.86                     8.33   
Denmark                              9.29                     8.33   
Canada                               9.64                     7.78   
Australia                            8.93                     7.78   
Switzerland                          9.29                     7.78   
Netherlands                          9.29                     8.33   
Luxembourg                           8.93                     6.67   
Germany                              8.57                     8.33   
United Kingdom                       7.50                     8.89   
Uruguay                              8.57                     6.11   
Austria                              7.86                     8.33   
Spain                                7.14                     7.78   
Mauritius                            8.21                     5.56   
Costa Rica                           7.50                     6.67   
France                               7.86                     7.78   

                Political culture  Civil liberties     Regime type  \
Country                                                              
Norway                      10.00             9.71  Full democracy   
Iceland                     10.00             9.71  Full democracy   
Sweden                      10.00             9.41  Full democracy   
New Zealand                  8.13            10.00  Full democracy   
Finland                      8.75             9.71  Full democracy   
Ireland                     10.00            10.00  Full democracy   
Denmark                      9.38             9.12  Full democracy   
Canada                       9.38             9.71  Full democracy   
Australia                    8.75            10.00  Full democracy   
Switzerland                  9.38             9.12  Full democracy   
Netherlands                  8.75             9.12  Full democracy   
Luxembourg                   8.75             9.71  Full democracy   
Germany                      7.50             9.41  Full democracy   
United Kingdom  

In [36]:
democracy_df.loc['Qatar'].shape

(9,)

### Democracy Index data clean-up and merger with donation dataset

In [12]:
# Pulling all unique foreign nations that made a donation to US universities

donation_countries = set(donations_df
                         .loc[:, 'Country of Giftor']
                         .str
                         .title())                      # Transform name of countries to title case

In [13]:
# Pulling all countries with a democracy rating

democracy_countries = set(democracy_df.index)

In [14]:
# Find countries that donated that does not appear in democracy index

intersect = donation_countries & democracy_countries  # Set intersection
donation_countries - intersect

{'Bahamas',
 'Barbados',
 'Bermuda',
 'Bosnia And Herzegovina',
 'Bosnia-Hercegovina',
 'British West Indies',
 'Burma',
 'Cayman Islands (The)',
 'Ceylon',
 'Curaçao',
 'England',
 'Grenada',
 'Guam',
 'Guernsey',
 'Isle Of Man',
 'Jersey',
 'Korea',
 'Liechtenstein',
 'Marshal Islands (The)',
 'Monaco',
 'Palestine, State Of',
 'Puerto Rico',
 'Scotland',
 'St. Lucia',
 'Swaziland',
 'Syrian Arab Republic',
 'The Netherlands',
 'Tokelau',
 'Trinidad',
 'United Nations',
 'Usa',
 'Virgin Islands (British)',
 'Wales'}

In [15]:
# Construct rename dictionary to correlate country names between the two datasets

countries_dict = {
 'Bahamas': 'Not Rated',                      # Countries that noted but not rated a Democracy Index are marked 'Not Rated'
 'Barbados': 'Not Rated',
 'Bermuda': 'United Kingdom',                 # Oversea territory of the UK
 'Bosnia And Herzegovina': 'Bosnia and Herzegovina',
 'Bosnia-Hercegovina': 'Bosnia and Herzegovina',
 'British West Indies': 'United Kingdom',     # Former British territory
 'Burma': 'Myanmar',
 'Cayman Islands (The)': 'United Kingdom',    # Oversea territory of the UK
 'Ceylon': 'Sri Lanka',                       # Part of the Netherlands
 'Curaçao': 'Netherlands',
 'England': 'United Kingdom',                 # Part of the UK
 'Grenada': 'Not Rated',
 'Guam': 'United States',                     # Territory of the US
 'Guernsey': 'United Kingdom',                # British Crown Dependency
 'Isle Of Man': 'United Kingdom',             # British Crown Dependency
 'Jersey': 'United Kingdom',                  # British Crown Dependency
 'Korea': 'South Korea',
 'Liechtenstein': 'Not Rated',
 'Marshal Islands (The)': 'Not Rated',
 'Monaco': 'Not Rated',
 'Palestine, State Of': 'Palestine',
 'Puerto Rico': 'United States',             # Commonwealth of the US
 'Scotland': 'United Kingdom',               # Part of the UK
 'St. Lucia': 'Not Rated',
 'Swaziland': 'Not Rated',
 'Syrian Arab Republic': 'Syria',
 'The Netherlands': 'Netherlands',
 'Tokelau': 'New Zealand',                   # Dependent territory of New Zealand
 'Trinidad': 'Trinidad and Tobago',
 'United Nations': 'Not Rated',
 'Usa': 'United States',
 'Virgin Islands (British)': 'United Kingdom',  # Oversea territory of the UK
 'Wales': 'United Kingdom',                  # Part of the UK
}

In [16]:
# Add column to foreign donation dataset for merging with Democracy Index dataset

donations_df.loc[:, 'Country to merge on'] = (     # Adding column
    donations_df.loc[:, 'Country of Giftor']       # Copy 'Country of Giftor' column
    .str
    .title())                                      # But transfor to title case

donations_df.replace(
    to_replace={'Country to merge on': countries_dict},   # Replace country name with countries_dict dictionary
    inplace=True)

merged_df = pd.merge(
    donations_df, 
    democracy_df, 
    how='left', 
    left_on='Country to merge on', 
    right_index=True)

merged_df.loc[:, 'Country of Giftor'] = (                # Transforming 'Country of Giftor' column to title case
    merged_df.loc[:, 'Country of Giftor']
    .str
    .title())

merged_df = (                                            # Setting up multi-index
    merged_df
    .set_index(['Country of Giftor', 'Institution Name'])
    .sort_index())

merged_df.head(20)

ID   OPEID  \
Country of Giftor Institution Name                                             
Afghanistan       Ball State University                          830  178600   
                  Johns Hopkins University                      1243  207700   
                  Johns Hopkins University                      2482  207700   
                  Johns Hopkins University                      2557  207700   
                  Johns Hopkins University                      6736  207700   
                  Johns Hopkins University                      7164  207700   
                  Johns Hopkins University                      8562  207700   
                  Johns Hopkins University                      9178  207700   
                  Johns Hopkins University                     12676  207700   
                  Ohio University                               7315  310000   
                  Ohio University                               7999  310000   
                  Ohio University                               9060  310000   
                  University of Illinois at Urbana-Champaign    4690  177500   
                  University of Illinois at Urbana-Champaign    6570  177500   
                  University of Illinois at Urbana-Champaign   19312  177500   
Angola            University of Oklahoma                        3675  318400   
Argentina         Columbia University in the City of New York   6335  270700   
                  Harvard University                            4579  215500   
                  Harvard University                            5979  215500   
                  Harvard University                            6448  215500   

                                                                    City  \
Country of Giftor Institution Name                                         
Afghanistan       Ball State University                           Muncie   
                  Johns Hopkins University                     Baltimore   
                  Johns Hopkins University                     Baltimore   
                  Johns Hopkins University                     Baltimore   
                  Johns Hopkins University                     Baltimore   
                  Johns Hopkins University                     Baltimore   
                  Johns Hopkins University                     Baltimore   
                  Johns Hopkins University                     Baltimore   
                  Johns Hopkins University                     Baltimore   
                  Ohio University                                 Athens   
                  Ohio University                                 Athens   
                  Ohio University                                 Athens   
                  University of Illinois at Urbana-Champaign   Champaign   
                  University of Illinois at Urbana-Champaign   Champaign   
                  University of Illinois at Urbana-Champaign   Champaign   
Angola            University of Oklahoma                          Norman   
Argentina         Columbia University in the City of New York   New York   
                  Harvard University                           Cambridge   
                  Harvard University                           Cambridge   
                  Harvard University                           Cambridge   

                                                              State  \
Country of Giftor Institution Name                                    
Afghanistan       Ball State University                          IN   
                  Johns Hopkins University                       MD   
                  Johns Hopkins University                       MD   
                  Johns Hopkins University                       MD   
                  Johns Hopkins University                       MD   
                  Johns Hopkins University                       MD   
                  Johns Hopkins University                      

In [19]:
merged_df.to_csv('donation_democracy_index.csv')

## Identifying schools receiving most donations from authoritarian (according to DI definition) countries

### Top 10 schools based on number of donations recived from authoritarian countries

In [20]:
authoritarian_df = (
    merged_df.
    loc[merged_df['Regime type']=='Authoritarian'])

top10schools_by_counts = (
    authoritarian_df.groupby('Institution Name')
    .count()
    .iloc[:, 0]
    .sort_values(ascending=False)
    .head(10))

top10schools_by_counts

Institution Name
University of California, Los Angeles          412
California Institute of Technology             346
Texas A&M University                           162
Stanford University                            159
Arizona State University                       157
Saint John's University                        155
Johns Hopkins University                       117
University of Michigan - Ann Arbor             106
Columbia University in the City of New York     96
Northwestern University                         89
Name: ID, dtype: int64

In [58]:
top10schools_by_counts.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area
ax.barh(top10schools_by_counts.index, top10schools_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations received')
ax.set_title('Top 10 universities based on number of donations recieved from authoritarian countries (2012-2018)')
for i, v in enumerate(top10schools_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)


# Note: this is donated contract + cash gifts

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

### Top 10 schools based on total dollars of donations recived from authoritarian countries

In [34]:
top10schools_by_dollars = (
    authoritarian_df.groupby('Institution Name')
    .sum().loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False)
    .head(10))
top10schools_by_dollars

dill.dump(top10schools_by_dollars, open('./top_10_schools_auth.pkd', 'wb'))

In [22]:
top10schools_by_dollars.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(10,5))                                 # Begin pyplot area
ax.barh(top10schools_by_dollars.index, top10schools_by_dollars.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million Dollars')
ax.set_title('Top 10 universities based on dollars of donations received from authoritarian countries')
for i, v in enumerate(top10schools_by_dollars):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)
plt.tight_layout()

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

### Top 10 schools based on total dollars of cash donations recived from authoritarian countries

In [61]:
top10schools_by_cash = (
    authoritarian_df.loc[authoritarian_df['Gift Type']=='Monetary Gift']
    .groupby('Institution Name')
    .sum()
    .loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False)
    .head(10))
top10schools_by_cash

Institution Name
Massachusetts Institute of Technology          190198291
Harvard University                              87979142
Stanford University                             75421948
New York University                             56297438
Johns Hopkins University                        53533107
Texas A&M University                            52364406
University of Pennsylvania                      35101067
Duke University                                 30564760
Yale University                                 24353520
Columbia University in the City of New York     21755763
Name: Foreign Gift Amount, dtype: int64

In [62]:
top10schools_by_cash.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                   # Initializing pyplot area
ax.barh(top10schools_by_cash.index, top10schools_by_cash.values*(10**-6), color='purple')    # Generating bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Top 10 universities based on dollars of cash donations received from authoritarian countries (2012-2018)')
for i, v in enumerate(top10schools_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

<ipython-input-62-5dc0f80de9e2>:2: RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).
  fig, ax = plt.subplots(figsize=(5,5))                                   # Initializing pyplot area


Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

## Identifying authoritarian countries making the most donations to US universities

### Top 10 authoritarian countries based on the number of donations made to US universities

In [63]:
top10countries_by_counts = (
    authoritarian_df.groupby('Country of Giftor')
    .count().iloc[:, 0]
    .sort_values(ascending=False)
    .head(10))

top10countries_by_counts

Country of Giftor
China                   1597
Saudi Arabia             719
Qatar                    444
United Arab Emirates     177
Kuwait                    83
Kazakhstan                54
Russia                    46
Vietnam                   43
Jordan                    22
Iraq                      18
Name: ID, dtype: int64

In [64]:
top10countries_by_counts.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(top10countries_by_counts.index, top10countries_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations made')
ax.set_title('Top 10 authoritarian countries based on number of donations made (2012-2018)')
for i, v in enumerate(top10countries_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)

<ipython-input-64-e2b07841c3f6>:2: RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).
  fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area


Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

### Top 10 authoritarian countries based on total donation dollar amount

In [23]:
top10countries_by_dollars = (
    authoritarian_df.groupby('Country of Giftor')
    .sum().loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False)
    .head(10))

top10countries_by_dollars

Country of Giftor
Qatar                   1478676069
China                    680273016
Saudi Arabia             649971586
United Arab Emirates     230945101
Russia                    99461804
Kuwait                    65782667
Kazakhstan                62179114
Iraq                      44276761
Vietnam                   11799413
Afghanistan                8891774
Name: Foreign Gift Amount, dtype: int64

In [24]:
top10countries_by_dollars.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(10,5))                                   # Initializing pyplot area
ax.barh(top10countries_by_dollars.index, top10countries_by_dollars.values*(10**-9), color='purple')    # Generating bar chart
ax.set_xlabel('Billion dollars')
ax.set_title('Top 10 authoritarian countries based on total donation dollar amount (2012-2018)')
for i, v in enumerate(top10countries_by_dollars):
    ax.text(v*(10**-9), i,'\$'+str('{:.2f}'.format(v*(10**-9))+'B'), color='black')
plt.margins(x=0.15)
plt.tight_layout()

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

### Top 10 authoritarian countries based on total cash donation dollar amount

In [25]:
top10countries_by_cash = (
    authoritarian_df.loc[authoritarian_df['Gift Type']=='Monetary Gift']
    .groupby('Country of Giftor')
    .sum().loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False)
    .head(10))

top10countries_by_cash

Country of Giftor
China                   323674776
Saudi Arabia            194561169
Russia                   95026310
Qatar                    74621261
United Arab Emirates     50152823
Kuwait                   45888708
Kazakhstan               36639149
Afghanistan               3890364
Burma                     2781683
Egypt                     2720885
Name: Foreign Gift Amount, dtype: int64

In [26]:
top10countries_by_cash.sort_values(inplace=True)

fig, ax = plt.subplots(figsize=(5,5))                                   # Initializing pyplot area
ax.barh(top10countries_by_cash.index, top10countries_by_cash.values*(10**-6), color='purple')    # Generating bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Top 10 authoritarian countries based on total cash donationn made (2012-2018)')
for i, v in enumerate(top10countries_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

#### Some quick observations:

* School receiving largest number of gifts from authoritarian countries: **University of California, Los Angeles** (UCLA)
* School receiving largest dollar amount of total gifts from authoritarian countries: **Carnegie Mellon University** (CMU)
* School receiving largest dollar amount of cash gifts from authoritarian countries: **Massachusetts Institute of Technology** (MIT)
* Authoritarian country donating the largest number of gifts: **China**
* Authoritarian country donating the largest dollar amount of total gifts: **Qatar**
* Authoritarian country donating the largest dollar amount of cash gifts: **China** (**Saudi Arabia** is 2<sup>nd</sup>)

## University of California, Los Angeles (UCLA)

UCLA recieved the most number of individual foreign donations (3,611) between 2012-2018. Out of those, 412 individual donations were from authoritarian countries (as defined by the Democracy Index), which made UCLA also the university recieving the most number of indivudual donations from authoritarian countries.

### Donation data from UCLA

In [27]:
UCLA_df = (
    merged_df.query('`Institution Name` == "University of California, Los Angeles"')   # Using pd.DataFrame.query
    .reset_index()                                                                     # Back quotes used to escape index
    .set_index(['Institution Name', 'Country of Giftor']))                             # labels with space

UCLA_df.head()

ID   OPEID  \
Institution Name                      Country of Giftor                  
University of California, Los Angeles Australia           1293  131500   
                                      Australia           5530  131500   
                                      Australia          12779  131500   
                                      Australia          13946  131500   
                                      Australia          14200  131500   

                                                                City State  \
Institution Name                      Country of Giftor                      
University of California, Los Angeles Australia          Los Angeles    CA   
                                      Australia          Los Angeles    CA   
                                      Australia          Los Angeles    CA   
                                      Australia          Los Angeles    CA   
                                      Australia          Los Angeles    CA   

                                                        Foreign Gift Received Date  \
Institution Name                      Country of Giftor                              
University of California, Los Angeles Australia                         2012-02-15   
                                      Australia                         2018-03-19   
                                      Australia                         2016-07-01   
                                      Australia                         2018-03-01   
                                      Australia                         2017-07-01   

                                                         Foreign Gift Amount  \
Institution Name                      Country of Giftor                        
University of California, Los Angeles Australia                      1255460   
                                      Australia                       313320   
                                      Australia                        21896   
                                      Australia                        10560   
                                      Australia                         9588   

                                                        Gift Type  \
Institution Name                      Country of Giftor             
University of California, Los Angeles Australia          Contract   
                                      Australia          Contract   
                                      Australia          Contract   
                                      Australia          Contract   
                                      Australia          Contract   

                                                                             Giftor Name  \
Institution Name                      Country of Giftor                                    
University of California, Los Angeles Australia                                      NaN   
                                      Australia          LOWY MEDICAL RESEARCH INSTITUTE   
                                      Australia                  UNIV OF NEW SOUTH WALES   
                                      Australia              Global Kinetics Corporation   
                                      Australia                   LOWY MEDICAL RESEARCH    

                                                        Country to merge on  \
Institution Name                      Country of Giftor                       
University of California, Los Angeles Australia                   Australia   
                                      Australia                   Australia   
                                      Australia                   Australia   
                                      Australia                   Australia   
                                      Australia                   Australia   

                                                         Rank  Score  \
Institution Name                      Country of Giftor                
Uni

#### List of foreign donations to UCLA grouped by countries

In [28]:
(UCLA_df.groupby('Country of Giftor')
 .count()
 .iloc[:,0]
 .sort_values(ascending=False))

Country of Giftor
England                 892
Japan                   535
Canada                  495
China                   394
Hong Kong               290
Switzerland             197
Germany                 159
Taiwan                  121
France                  117
Australia                83
South Korea              73
Spain                    61
Israel                   60
Korea                    28
Singapore                19
The Netherlands          17
Saudi Arabia             13
Italy                    10
Norway                    7
Ireland                   6
South Africa              6
Brazil                    5
Greece                    4
Malaysia                  4
India                     3
Denmark                   3
United Arab Emirates      3
Bahamas                   2
Eritrea                   1
Ecuador                   1
Russia                    1
Ceylon                    1
Name: ID, dtype: int64

#### Total dollars donated to UCLA by each country

In [29]:
(UCLA_df.groupby('Country of Giftor')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False))

Country of Giftor
England                 51186060
Japan                   23878716
Hong Kong               16606114
Switzerland             14286549
China                   13784326
Saudi Arabia            12032553
Germany                 11816638
France                   5884617
Canada                   5529592
Israel                   5334192
Korea                    4147817
Spain                    3532947
Taiwan                   3470583
Australia                2223854
The Netherlands          2068299
Ireland                  1887475
South Africa             1824427
South Korea              1715667
Denmark                  1072162
India                     990140
Russia                    673000
Greece                    539275
United Arab Emirates      453438
Singapore                 410505
Italy                     371846
Bahamas                   299983
Norway                    289839
Brazil                    276205
Malaysia                  250625
Ceylon                   

#### Total cash only donations made to UCLA by each country

In [30]:
(UCLA_df.loc[UCLA_df['Gift Type']=='Monetary Gift']
 .groupby('Country of Giftor')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False))

Country of Giftor
England                 27045805
Hong Kong               11326545
Japan                    9526275
China                    5814723
Spain                    3329965
Taiwan                   3105107
Switzerland              1693504
Canada                   1368880
South Korea              1104869
The Netherlands           590691
Greece                    539275
France                    483368
Singapore                 410505
Korea                     369230
Germany                   304694
Norway                    289839
Brazil                    276205
Malaysia                  250625
Saudi Arabia              210620
Israel                    112939
Bahamas                    99983
Australia                  17785
Italy                       4425
Ireland                     1100
United Arab Emirates        1021
India                        140
South Africa                 121
Ceylon                       100
Ecuador                       21
Eritrea                  

#### List of donations made to UCLA by authoritarian countries

In [31]:
UCLA_by_counts = (
    UCLA_df.loc[UCLA_df['Regime type']=='Authoritarian']
    .groupby('Country of Giftor')
    .count()
    .iloc[:,0].sort_values(ascending=False))

UCLA_by_counts

Country of Giftor
China                   394
Saudi Arabia             13
United Arab Emirates      3
Russia                    1
Eritrea                   1
Name: ID, dtype: int64

In [32]:
UCLA_by_counts.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(UCLA_by_counts.index, UCLA_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations made')
ax.set_title('Number of donations made to UCLA by authoritarian countries (2012-2018)')
for i, v in enumerate(UCLA_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

#### Total donation dollars received by UCLA from authoritarian countries

In [33]:
UCLA_by_dollars = (
    UCLA_df.loc[UCLA_df['Regime type']=='Authoritarian']
    .groupby('Country of Giftor')
    .sum()
    .loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False))

UCLA_by_dollars

Country of Giftor
China                   13784326
Saudi Arabia            12032553
Russia                    673000
United Arab Emirates      453438
Eritrea                        3
Name: Foreign Gift Amount, dtype: int64

In [34]:
UCLA_by_dollars.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(UCLA_by_dollars.index, UCLA_by_dollars.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total donation accepted by UCLA from authoritarian countries (2012-2018)')
for i, v in enumerate(UCLA_by_dollars):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

#### Total cash donation accepted by UCLA from authoritarian countries

In [35]:
UCLA_by_cash = (
    UCLA_df.loc[(UCLA_df['Regime type']=='Authoritarian') & (UCLA_df['Gift Type']=='Monetary Gift')]
    .groupby('Country of Giftor')
    .sum().loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False))

UCLA_by_cash

Country of Giftor
China                   5814723
Saudi Arabia             210620
United Arab Emirates       1021
Eritrea                       3
Name: Foreign Gift Amount, dtype: int64

In [36]:
UCLA_by_cash.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(UCLA_by_cash.index, UCLA_by_cash*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total cash donation accepted by UCLA from authoritarian countries (2012-2018)')
for i, v in enumerate(UCLA_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

## Carnegie Mellon University (CMU)

UCLA recieved the most donations in dollar amount from authoritarian countries in 2012-2018. These donations include contracts and cash. However, CMU was not within the top ten list of receiving the most individual donations from authoritarian countries. This implies that CMU accepted fewer donations from these countries, but each donations were of significantly higher dollar amount.

### Donation data from CMU

In [23]:
CMU_df = (
    merged_df.query('`Institution Name` == "Carnegie Mellon University"')   # Using pd.DataFrame.query
    .reset_index()                                                                     # Back quotes used to escape index
    .set_index(['Institution Name', 'Country of Giftor']))                             # labels with space

CMU_df.head()

ID   OPEID        City State  \
Institution Name           Country of Giftor                                   
Carnegie Mellon University Bermuda            5717  324200  Pittsburgh    PA   
                           Bermuda            5718  324200  Pittsburgh    PA   
                           Bermuda            5719  324200  Pittsburgh    PA   
                           Bermuda            5720  324200  Pittsburgh    PA   
                           Bermuda            5721  324200  Pittsburgh    PA   

                                             Foreign Gift Received Date  \
Institution Name           Country of Giftor                              
Carnegie Mellon University Bermuda                           2016-02-16   
                           Bermuda                           2016-02-16   
                           Bermuda                           2016-02-16   
                           Bermuda                           2016-02-16   
                           Bermuda                           2016-02-16   

                                              Foreign Gift Amount Gift Type  \
Institution Name           Country of Giftor                                  
Carnegie Mellon University Bermuda                       75000000  Contract   
                           Bermuda                       75000000  Contract   
                           Bermuda                       75000000  Contract   
                           Bermuda                       75000000  Contract   
                           Bermuda                       75000000  Contract   

                                             Giftor Name Country to merge on  \
Institution Name           Country of Giftor                                   
Carnegie Mellon University Bermuda                   NaN      United Kingdom   
                           Bermuda                   NaN      United Kingdom   
                           Bermuda                   NaN      United Kingdom   
                           Bermuda                   NaN      United Kingdom   
                           Bermuda                   NaN      United Kingdom   

                                              Rank  Score  \
Institution Name           Country of Giftor                
Carnegie Mellon University Bermuda            14.0   8.52   
                           Bermuda            14.0   8.52   
                           Bermuda            14.0   8.52   
                           Bermuda            14.0   8.52   
                           Bermuda            14.0   8.52   

                                              Electoral process and pluralism  \
Institution Name           Country of Giftor                                    
Carnegie Mellon University Bermuda                                       9.58   
                           Bermuda                                       9.58   
                           Bermuda                                       9.58   
                           Bermuda                                       9.58   
                           Bermuda                                       9.58   

                                              Functioning of government  \
Institution Name           Country of Giftor                              
Carnegie Mellon University Bermuda                                  7.5   
                           Bermuda                                  7.5   
                           Bermuda                                  7.5   
                           Bermuda                                  7.5   
                           Bermuda                                  7.5   

                                              Political participation  \
Institution Name           Country of Giftor                            
Carnegie Mellon University Bermuda                               8.89   
                           Bermuda                               8.89   
                           Bermu

#### List of foreign donations to CMU grouped by countries

In [24]:
(CMU_df.groupby('Country of Giftor')
 .count()
 .iloc[:,0]
 .sort_values(ascending=False))

Country of Giftor
Japan                   25
Qatar                   22
South Korea             19
India                   16
Bermuda                 10
England                  8
Switzerland              7
Taiwan                   6
China                    6
The Netherlands          4
Portugal                 4
Russia                   4
France                   3
Colombia                 3
South Africa             3
Canada                   3
Hong Kong                3
Chile                    2
Singapore                2
Germany                  2
Czech Republic           1
British West Indies      1
United Arab Emirates     1
Italy                    1
Korea                    1
Norway                   1
Rwanda                   1
Saudi Arabia             1
Sweden                   1
Thailand                 1
Mexico                   1
Name: ID, dtype: int64

#### Total dollars donated to CMU by each country

In [25]:
(CMU_df.groupby('Country of Giftor')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False))

Country of Giftor
Bermuda                 750000000
Qatar                   376325840
Thailand                 67700000
Portugal                 37492100
India                    29399473
Japan                    21851301
Canada                   11937151
Hong Kong                11250000
South Korea               9822307
Taiwan                    8150000
Switzerland               8093900
South Africa              7940779
United Arab Emirates      5000000
England                   4290688
Singapore                 4165000
China                     4146500
Colombia                  3792101
Russia                    2860500
Germany                   2795576
Chile                     2583435
The Netherlands           2383856
France                    1088287
Saudi Arabia              1050000
Korea                      964000
Italy                      720000
Sweden                     617566
Norway                     383990
Czech Republic             317520
Mexico                     307

#### Total cash only donations made to CMU by each country

In [26]:
(CMU_df.loc[CMU_df['Gift Type']=='Monetary Gift']
 .groupby('Country of Giftor')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False))

Country of Giftor
India             16650000
Canada            11687151
Qatar              5456800
Switzerland        4800000
Japan              3637243
China              2500000
England            2029602
Hong Kong          1250000
Czech Republic      317520
Rwanda              279334
Name: Foreign Gift Amount, dtype: int64

(t_stat, p_value, f'Different={p_value<0.01}')

In [41]:
CMU_by_counts = (
    CMU_df.loc[CMU_df['Regime type']=='Authoritarian']
    .groupby('Country of Giftor')
    .count()
    .iloc[:,0].sort_values(ascending=False))

CMU_by_countsy_counts

NameError: name 'CMU_by_countsy_counts' is not defined

In [ ]:
CMU_by_counts.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(CMU_by_counts.index, CMU_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations made')
ax.set_title('Number of donations made to CMU by authoritarian countries (2012-2018)')
for i, v in enumerate(CMU_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)

#### Total donation dollars received by CMU from authoritarian countries

In [27]:
CMU_by_dollars = (
    CMU_df.loc[CMU_df['Regime type']=='Authoritarian']
    .groupby('Country of Giftor')
    .sum().loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False))

CMU_by_dollars

Country of Giftor
Qatar                   376325840
United Arab Emirates      5000000
China                     4146500
Russia                    2860500
Saudi Arabia              1050000
Rwanda                     279334
Name: Foreign Gift Amount, dtype: int64

In [33]:
CMU_by_dollars.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(10, 3))                                 # Begin pyplot area

ax.barh(CMU_by_dollars.index, CMU_by_dollars.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total donation accepted by CMU from authoritarian countries (2012-2018)')
for i, v in enumerate(CMU_by_dollars):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)
plt.tight_layout()

Canvas(toolbar=Toolbar(toolitems=[('Home', 'Reset original view', 'home', 'home'), ('Back', 'Back to previous …

#### Total cash donation accepted by CMU from authoritarian countries

In [ ]:
CMU_by_cash = (
    CMU_df.loc[(CMU_df['Regime type']=='Authoritarian') & (CMU_df['Gift Type']=='Monetary Gift')]
    .groupby('Country of Giftor')
    .sum().loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False))

CMU_by_cash

In [ ]:
CMU_by_cash.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(CMU_by_cash.index, CMU_by_cash.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total cash donation accepted by CMU from authoritarian countries (2012-2018)')
for i, v in enumerate(CMU_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

## Massachusetts Institute of Technology (MIT)

If only cash donations were being measured, then MIT received the most cash donations from authoritarian countries. Without a full analysis we can't be sure, but we can see from CMU that often foreign countries would make a massive donation in the form of a contract. These type of donations generally were much higher in dolalr value than purely cash donations.

#### List of foreign donations to MIT grouped by countries

In [ ]:
(MIT_df.groupby('Country of Giftor')
 .count()
 .iloc[:,0]
 .sort_values(ascending=False))

#### Total dollars donated to MIT by each country

In [ ]:
(MIT_df.groupby('Country of Giftor')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False))

#### Total cash donated to MIT by each country

In [ ]:
(MIT_df.loc[MIT_df['Gift Type']=='Monetary Gift']
 .groupby('Country of Giftor')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False))

#### List of donations made to MIT by authoritarian countries

In [ ]:
MIT_by_counts = (MIT_df.loc[MIT_df['Regime type']=='Authoritarian']
 .groupby('Country of Giftor')
 .count()
 .iloc[:,0].sort_values(ascending=False))

MIT_by_counts

In [ ]:
MIT_by_counts.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(MIT_by_counts.index, MIT_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations made')
ax.set_title('Number of donations made to MIT by authoritarian countries (2012-2018)')
for i, v in enumerate(MIT_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)


#### Total donation dollars received by MIT from authoritarian countries

In [ ]:
MIT_by_dollars = (MIT_df.loc[MIT_df['Regime type']=='Authoritarian']
 .groupby('Country of Giftor')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False))

MIT_by_dollars

In [ ]:
MIT_by_dollars.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(MIT_by_dollars.index, MIT_by_dollars.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total donation accepted by MIT from authoritarian countries (2012-2018)')
for i, v in enumerate(MIT_by_dollars):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

#### Total cash donation accepted by MIT from authoritarian countries

In [ ]:
MIT_by_cash = (MIT_df.loc[(MIT_df['Regime type']=='Authoritarian') & (MIT_df['Gift Type']=='Monetary Gift')]
 .groupby('Country of Giftor')
 .sum().loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False))

MIT_by_cash

In [ ]:
MIT_by_cash.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(MIT_by_cash.index, MIT_by_cash.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total cash donation accepted by MIT from authoritarian countries (2012-2018)')
for i, v in enumerate(MIT_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

### People's Republic of China

China was ranked as the authoritarian country that made the most number of individual donations and also the hightest amount of total cash donations between 2012 to 2018.

#### List of donations made by China to US universities

In [ ]:
china_df = authoritarian_df.loc['China'].copy()
china_df.head()

#### The 10 US universitites China donated to the most number of times

In [ ]:
china_by_counts = (china_df.groupby('Institution Name')
 .count()
 .iloc[:,0]
 .sort_values(ascending=False)
 .head(10))

china_by_counts

In [ ]:
china_by_counts.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(china_by_counts.index, china_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations made')
ax.set_title('Number of donations made by China to US universities (2012-2018)')
for i, v in enumerate(china_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)

#### The 10 US universitites China donated most amount of total dollars

In [ ]:
china_by_dollars = (china_df.groupby('Institution Name')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False)
 .head(10))

china_by_dollars

In [ ]:
china_by_dollars.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(china_by_dollars.index, china_by_dollars.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total donation donated by China to US universities (2012-2018)')
for i, v in enumerate(china_by_dollars):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

#### The 10 US universitites China donated most amount of total cash

In [ ]:
china_by_cash = (china_df.loc[china_df['Gift Type']=='Monetary Gift']
 .groupby('Institution Name')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False)
 .head(10))

china_by_cash

In [ ]:
china_by_cash.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(china_by_cash.index, china_by_cash.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total cash donation donated by China to US universities (2012-2018)')
for i, v in enumerate(china_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

### State of Qatar

Qatar was ranked as the authoritarian country that made the highest total dollar amount of donations to US universities between 2012 to 2018. This was primarily due to the large contract-based donations Qatar made, as China was the authoritarian country that made the highest cash donations, followed by Saudi Arabia.

#### List of donations made by Qatar to US universities

In [ ]:
qatar_df = authoritarian_df.loc['Qatar'].copy()
qatar_df.head()

#### The 10 US universitites Qatar donated to the most number of times

In [ ]:
qatar_by_counts = (qatar_df.groupby('Institution Name')
 .count()
 .iloc[:,0]
 .sort_values(ascending=False)
 .head(10))

qatar_by_counts

In [ ]:
qatar_by_counts.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(qatar_by_counts.index, qatar_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations made')
ax.set_title('Number of donations made by Qatar to US universities (2012-2018)')
for i, v in enumerate(qatar_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)


#### The 10 US universitites Qatar donated most amount of total dollars

In [ ]:
qatar_by_dollars = (qatar_df.groupby('Institution Name')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False)
 .head(10))

qatar_by_dollars

In [ ]:
qatar_by_dollars.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(qatar_by_dollars.index, qatar_by_dollars.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total donation donated by Qatar to US universities (2012-2018)')
for i, v in enumerate(qatar_by_dollars):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

#### The 10 US universitites Qatar donated most amount of total cash

In [ ]:
qatar_by_cash = (qatar_df.loc[qatar_df['Gift Type']=='Monetary Gift']
 .groupby('Institution Name')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False)
 .head(10))

qatar_by_cash

In [ ]:
qatar_by_cash.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(qatar_by_cash.index, qatar_by_cash.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total cash donation donated by Qatar to US universities (2012-2018)')
for i, v in enumerate(qatar_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

### Kingdom of Saudi Arabia

Saudi Arabia only ranked as the 2nd authoritarian country that donated the most amount of cash to US university, as China was ranked the 1st. We have shown analyses of China above, below is of Saudi Arabia.

#### List of donations made by Saudi Arabia to US universities

In [ ]:
saudi_df = authoritarian_df.loc['Saudi Arabia'].copy()
saudi_df.head()

#### The 10 US universitites Saudi Arabia donated to the most number of times

In [ ]:
saudi_by_counts = (saudi_df.groupby('Institution Name')
 .count()
 .iloc[:,0]
 .sort_values(ascending=False)
 .head(10))

saudi_by_counts

In [ ]:
saudi_by_counts.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(saudi_by_counts.index, saudi_by_counts.values, color='purple')    # Generate basic bar chart
ax.set_xlabel('Number of donations made')
ax.set_title('Number of donations made by Saudi Arabia to US universities (2012-2018)')
for i, v in enumerate(saudi_by_counts.values):           # Add numeric labels to bars for clarity
     ax.text(v, i, str(v), color='black')
plt.margins(x=0.1)

#### The 10 US universitites Saudi Arabia donated most amount of total dollars

In [ ]:
saudi_by_dollars = (saudi_df.groupby('Institution Name')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False)
 .head(10))

saudi_by_dollars

In [ ]:
saudi_by_dollars.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(saudi_by_dollars.index, saudi_by_dollars.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total donation donated by Saudi Arabia to US universities (2012-2018)')
for i, v in enumerate(saudi_by_dollars):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

#### The 10 US universitites Saudi Arabia donated most amount of total cash

In [ ]:
saudi_by_cash = (saudi_df.loc[saudi_df['Gift Type']=='Monetary Gift']
 .groupby('Institution Name')
 .sum()
 .loc[:, 'Foreign Gift Amount']
 .sort_values(ascending=False)
 .head(10))

saudi_by_cash

In [ ]:
saudi_by_cash.sort_values(inplace=True)
fig, ax = plt.subplots(figsize=(5,5))                                 # Begin pyplot area

ax.barh(saudi_by_cash.index, saudi_by_cash.values*(10**-6), color='purple')    # Generate basic bar chart
ax.set_xlabel('Million dollars')
ax.set_title('Total cash donation donated by Saudi Arabia to US universities (2012-2018)')
for i, v in enumerate(saudi_by_cash):
    ax.text(v*(10**-6), i,'\$'+str('{:.2f}'.format(v*(10**-6))+'M'), color='black')
plt.margins(x=0.25)

#### Summary foreign donation information

Below are the summary foreign donation information of the top 50 US universities sorted by number of individual foreign donations recieved

In [ ]:
most_number = (
    merged_df.groupby('Institution Name')
    .count()
    .iloc[:, 0]
    .sort_values(ascending=False)
    .copy())

most_dollars = (
    merged_df.groupby('Institution Name')
    .sum()
    .loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False)
    .copy())

most_cash = (
    merged_df.loc[merged_df['Gift Type']=='Monetary Gift']
    .groupby('Institution Name')
    .sum()
    .loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False)
    .copy())

simplified_schools = pd.DataFrame(most_number).rename({'ID': 'Number of donations received'}, axis=1)
simplified_schools.loc[:, 'Total dollars received (including contracts)'] = most_dollars
simplified_schools.loc[:, 'Total cash received'] = most_cash

simplified_schools.head(50)

Below are the summary foreign donation information of the top 50 countries sorted by number of individual donations made

In [ ]:
most_number_c = (
    merged_df.groupby('Country of Giftor')
    .count()
    .iloc[:, 0]
    .sort_values(ascending=False)
    .copy())

most_dollars_c = (
    merged_df.groupby('Country of Giftor')
    .sum()
    .loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False)
    .copy())

most_cash_c = (
    merged_df.loc[merged_df['Gift Type']=='Monetary Gift']
    .groupby('Country of Giftor')
    .sum()
    .loc[:, 'Foreign Gift Amount']
    .sort_values(ascending=False)
    .copy())

simplified_countries = pd.DataFrame(most_number_c).rename({'ID': 'Number of donations received'}, axis=1)
simplified_countries.loc[:, 'Total dollars received (including contracts)'] = most_dollars_c
simplified_countries.loc[:, 'Total cash received'] = most_cash_c
simplified_countries.columns = ['Number of donations given', 'Total dollars given', 'Total cash given']

simplified_countries.head(50)

#### Statistical analyses

Here we consider the correlation (via Pearson's correlation coefficient) between the number of donation recieved, total dollars received, and total cash received by each school.

In [ ]:
simplified_schools.corr()

In [ ]:
fig, ax = plt.subplots(figsize=(5,5))                                  
ax.scatter(
    simplified_schools.loc[:, 'Number of donations received'], 
    simplified_schools.loc[:, 'Total dollars received (including contracts)']*(10**-9),
    color='purple',
)
ax.set_xlabel('Number of donations')
ax.set_ylabel('Billion Dollars')
ax.set_title('Correlation graph between number of donations received and total dollars received by each US university')

Based on the Pearson's correlation coefficient and visual observation of the correlation graph, there is little correlation between the number of donations received by a university and the total dollar amount it received.

Here we consider the correlation (via Pearson's correlation coefficient) between the number of donation made, total dollars donated, and total cash donated by each country.

In [ ]:
simplified_countries.corr()

In [ ]:
fig, ax = plt.subplots(figsize=(5,5))                                  
ax.scatter(
    simplified_countries.loc[:, 'Number of donations given'], 
    simplified_countries.loc[:, 'Total dollars given']*(10**-9),
    color='purple')
ax.set_xlabel('Number of donations')
ax.set_ylabel('Billion Dollars')
ax.set_title('Correlation graph between number of donations given and total dollars given by each country')

The correlation between number of donations given by each countries and the total dollar amount (including contracts) it gave is relatively strong, according to Pearson's correlation coefficient and visual observation of the correlation graph.